### PR TITLE
Reland 39919 CupertinoPicker

### PR DESCRIPTION
**Pull request by LongCatIsLooong**
## Description

Reland a reverted breaking change #39919.

/cc @liyuqian.

## Related Issues

 https://github.com/flutter/flutter/issues/41654

## Tests

I added the following tests:

*Replace this with a list of the tests that you added as part of this PR. A change in behaviour with no test covering it
will likely get reverted accidentally sooner or later. PRs must include tests for all changed/updated/fixed behaviors. See [Test Coverage].*

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [ ] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: *Replace with a link to your migration guide*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/rendering/list_wheel_viewport.dart [hunk 1]
```diff
@@ -822,20 +845,16 @@ class RenderListWheelViewport
 
     // Offset that helps painting everything in the center (e.g. angle = 0).
     final Offset offsetToCenter = Offset(
-        untransformedPaintingCoordinates.dx,
-        -_topScrollMarginExtent);
+      untransformedPaintingCoordinates.dx,
+      -_topScrollMarginExtent
```
Comment (justinmc) [inline packages/flutter/lib/src/rendering/list_wheel_viewport.dart]: Comma here at the end of the line.

File: packages/flutter/lib/src/rendering/list_wheel_viewport.dart [hunk 2]
```diff
@@ -927,20 +944,26 @@ class RenderListWheelViewport
     Matrix4 cylindricalTransform,
     Offset offsetToCenter,
   ) {
+    // Paint child with cylindrically.
```
Comment (justinmc) [inline packages/flutter/lib/src/rendering/list_wheel_viewport.dart]: Nit: Should this just be "Paint child cylindrically."?  Or does it mean to paint it with the "cylindrically" method?  Also see line 956 below.

**Review by justinmc — APPROVED** [review]
LGTM, just 2 nitpicks.

Should this be marked as a breaking change?

File: packages/flutter/lib/src/rendering/list_wheel_viewport.dart (hunk 2 above)
Comment (LongCatIsLooong) [reply packages/flutter/lib/src/rendering/list_wheel_viewport.dart]: Forgot to delete it I guess.

**Comment by Piinks** [comment]
FYI @kjlubick it looks like the triaged tryjob results did not carry over when this landed on master. They appeared under By Blame for triage after landing, and fired off test failures on other PRs. I am going to see if I can reproduce.

**Pull request merged**